packages/sdk/src/services/chainManager.ts

```
@@ -0,0 +1,77 @@
+import { createPublicClient, extractChain, http, type PublicClient } from 'viem'
+
+import { SUPPORTED_CHAINS } from '../constants/supportedChains.js'
+import type { ChainConfig } from '../types/verbs.js'
+
+/**
+ * Chain Manager Service
+ * @description Manages public clients and chain infrastructure for the Verbs SDK
+ */
+export class ChainManager {
+  private publicClients: Map<number, PublicClient>
+  private chainConfigs: ChainConfig[]
+
+  constructor(chains: ChainConfig[]) {
+    this.chainConfigs = chains
+    this.publicClients = this.createPublicClients(chains)
+  }
+
+  /**
+   * Create public clients for all configured chains
+   */
+  private createPublicClients(
+    chains: ChainConfig[],
+  ): Map<number, PublicClient> {
+    const clients = new Map<number, PublicClient>()
+
+    for (const chainConfig of chains) {
+      const client = createPublicClient({
+        chain: SUPPORTED_CHAINS.find(chain => chain.id === chainConfig.chainId),
+        transport: http(chainConfig.rpcUrl),
+      })
+
+      clients.set(chainConfig.chainId, client)
+    }
```

🤔 I think we should throw if an unsupported chain is passed in

---

updated!